src/main/java/com/datatorrent/netlet/DefaultEventLoop.java

```diff
@@ -76,8 +78,14 @@ public DefaultEventLoop(String id) throws IOException
   public synchronized Thread start()
   {
     logger.debug("Starting {}", this);
-    if (++refCount == 1) {
+    refCount++;
+    if (eventThread == null || !eventThread.isAlive()) {
+      if (refCount > 1) {
+        logger.warn("Restarting previously terminated event loop {} thread {}", this, eventThread);
+      }
       eventThread = new Thread(this, id);
+      eventThread.setPriority(priority);
+      eventThread.setDaemon(true);
```

Comment:
It is more a fallback to prevent event loop thread from preventing JVM to exit. It should be still possible to join on the thread and wait for it to terminate.